File: Smart Bee Email Interface/src/types/insight.ts
export interface SuggestedAction {
    action_id: string;
    action_type: string;
    payload: Record<string, any>;
    status: string;
  }
  
  export interface Insight {
    email_id: string;
    summary: string;
    intent: string;
    priority: string;
    confidence: number;
    actions: SuggestedAction[];
  }

  

File: smart_bee_backend_starter/app/schemas/insight.py
from pydantic import BaseModel
from typing import List, Dict, Any, Optional
from datetime import datetime
from app.schemas.action import SuggestedActionResponse

class InsightResponse(BaseModel):
    email_id: str
    sender: str
    subject: Optional[str]
    received_at: datetime
    category: Optional[str]
    summary: str
    intent: str
    priority: str
    confidence: float
    entities: Dict[str, Any]
    actions: List[SuggestedActionResponse]
    rationale: Optional[str] = None
